### PR TITLE
[DOM-50802] Provider produced inconsistent result after apply 

### PR DESCRIPTION
**Pull request by ldebello-ddl**
When creating a configuration that uses a schema with secret semantics the provider was failing.

# Root Cause
Terraform supports provided or computed fields, we were handling the "definition" field as a mix of the two, because we use the provided value but override with the computed value in the response.

# Changes

* Change the "definition" field to sensitive to avoid being shown in the plan and output
* Upgrade testing library
* Change Read to resolve secrets to avoid secrets:// prefix when resolving
* Fixing some linter errors
* Changing method from method pointer (I don't remember the exact reason but I am almost sure a warning from IDE when creating a test)
* Copying scope for schemas because they were missed so every upgrade was triggering a recreation

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -37,16 +41,16 @@ type RavelProviderModel struct {
...
 	resp.Version = p.version
 }
 
-func (p *RavelProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
+func (p RavelProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
 	resp.Schema = schema.Schema{
 		Attributes: map[string]schema.Attribute{
 			"url": schema.StringAttribute{
-				Description: fmt.Sprintf("Host URL for Ravel. Can be defined from env var RAVEL_URL"),
+				Description: "Host URL for Ravel. Can be defined from env var RAVEL_URL",
```
Comment (ddl-fpoblete) [inline internal/provider/provider.go]:
```suggestion
				Description: fmt.Sprintf("Host URL for Ravel. Can be defined from env var %s", RavelURL),
```

File: internal/provider/provider.go [hunk 2]
```diff
@@ -20,6 +19,11 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
+const (
+	RavelToken = "RAVEL_TOKEN"
+	RavelURL   = "RAVEL_URL"
```
Comment (ddl-fpoblete) [inline internal/provider/provider.go]:
```suggestion
	EnvRavelToken = "RAVEL_TOKEN"
	EnvRavelURL   = "RAVEL_URL"
```

File: internal/resources/configuration_resource_test.go [hunk 3]
```diff
@@ -1,52 +1,135 @@
...
+            }
+        }
+    }
+}`
+
+func mockAPIHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+
+	if r.Method == "GET" {
```
Comment (ddl-fpoblete) [inline internal/resources/configuration_resource_test.go]:
```suggestion
	if r.Method == http.MethodGet {
```

File: internal/resources/configuration_resource_test.go [hunk 4]
```diff
@@ -1,52 +1,135 @@
...
+		_, err := w.Write([]byte(getAPIResponse))
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+
+	} else if r.Method == "POST" {
```
Comment (ddl-fpoblete) [inline internal/resources/configuration_resource_test.go]:
```suggestion
	} else if r.Method == http.MethodPost {
```

File: internal/resources/configuration_resource.go [hunk 5]
```diff
@@ -271,6 +265,8 @@ func (r *ConfigurationResource) Read(ctx context.Context, req resource.ReadReque
 		schema.Name = types.StringValue(configuration.Spec.ConfigurationFormat.Name)
 		schema.Version = types.StringValue(configuration.Spec.ConfigurationFormat.Version)
 
+		schema.Scope = data.Schema.Scope
```
Comment (ddl-fpoblete) [inline internal/resources/configuration_resource.go]: I think we should copy the scope from `configuration.Spec.ConfigurationFormat.Scope` here

File: internal/provider/provider.go (hunk 2 above)
Comment (ldebello-ddl) [reply internal/provider/provider.go]: fixed

File: internal/provider/provider.go (hunk 1 above)
Comment (ldebello-ddl) [reply internal/provider/provider.go]: fixed

File: internal/resources/configuration_resource.go (hunk 5 above)
Comment (ldebello-ddl) [reply internal/resources/configuration_resource.go]: fixed

File: internal/resources/configuration_resource_test.go (hunk 3 above)
Comment (ldebello-ddl) [reply internal/resources/configuration_resource_test.go]: fixed

File: internal/resources/configuration_resource_test.go (hunk 4 above)
Comment (ldebello-ddl) [reply internal/resources/configuration_resource_test.go]: fixed

**Review by ddl-fpoblete — APPROVED** [review]

File: it/stmp_configuration/main.tf [hunk 6]
```diff
@@ -0,0 +1,37 @@
+resource "ravel_configuration" "stmp_configuration" {
```
Comment (ddl-dclegg) [inline it/stmp_configuration/main.tf]: Nit: I'm guessing this (and the filename) were supposed to be `smtp` instead of `stmp`, right?

**Comment by ddl-dclegg** [comment]
Is it intentional for us to be [building](https://github.com/dominodatalab/terraform-provider-ravel/blob/ad31d20504a2b43e666800343eab13f95ee24237/go.mod#L1) this as `github.com/cerebrotech/terraform-provider-ravel`, not `github.com/dominodatalab/terraform-provider-ravel`?

File: it/stmp_configuration/main.tf (hunk 6 above)
Comment (ldebello-ddl) [reply it/stmp_configuration/main.tf]: thanks, fixed

**Comment by ldebello-ddl** [comment]
> Is it intentional for us to be [building](https://github.com/dominodatalab/terraform-provider-ravel/blob/ad31d20504a2b43e666800343eab13f95ee24237/go.mod#L1) this as `github.com/cerebrotech/terraform-provider-ravel`, not `github.com/dominodatalab/terraform-provider-ravel`?

we realized the repo should be public to be published in terraform registry, so probably we create in the other org but forget to change that, I will change it as part of another PR, because not totally sure if that could break something

File: it/stmp_configuration/main.tf (hunk 6 above)
Comment (ddl-dclegg) [reply it/stmp_configuration/main.tf]: Thanks for catching the file name.  Can we update the resource too?

Comment (ldebello-ddl) [reply it/stmp_configuration/main.tf]: thanks, I didn't noticed that one :)

**Review by ddl-dclegg — APPROVED** [review]

**Review by domino-pablo-rodriguez — APPROVED** [review]
lgtm

**Pull request merged**